e2e-tests/native/test/helpers/Gestures.ts

```diff
@@ -0,0 +1,179 @@
+   * check if it visible after x amount of scrolls
+   */
+  static async checkIfDisplayedWithSwipeUp(
+    element: ChainablePromiseElement<Promise<WebdriverIO.Element>>,
+    maxScrolls: number,
+    amount = 0
+  ): Promise<void> {
+    // If the element is not displayed and we haven't scrolled the max amount of scrolls
+    // then scroll and execute the method again
+    if (!(await element.isDisplayed()) && amount <= maxScrolls) {
+      await this.swipeUp(SWIPE_PERCENTAGE)
+      await this.checkIfDisplayedWithSwipeUp(element, maxScrolls, amount + 1)
+    } else if (amount > maxScrolls) {
+      // If the element is still not visible after the max amount of scroll let it fail
+      throw new Error(`The element '${element}' could not be found or is not visible.`)
+    }
+
+    // The element was found, proceed with the next action
+  }
+
+  /**
+   * Swipe down based on a percentage
+   */
+  static async swipeDown(percentage = 1): Promise<void> {
+    await this.swipeOnPercentage(
+      this.calculateXY(SWIPE_DIRECTION.down.start, percentage),
+      this.calculateXY(SWIPE_DIRECTION.down.end, percentage)
+    )
+  }
+
+  /**
+   * Swipe Up based on a percentage
+   */
+  static async swipeUp(percentage = 1): Promise<void> {
+    await this.swipeOnPercentage(
+      this.calculateXY(SWIPE_DIRECTION.up.start, percentage),
+      this.calculateXY(SWIPE_DIRECTION.up.end, percentage)
+    )
+  }
+
+  /**
+   * Swipe left based on a percentage
+   */
+  static async swipeLeft(percentage = 1): Promise<void> {
+    await this.swipeOnPercentage(
+      this.calculateXY(SWIPE_DIRECTION.left.start, percentage),
+      this.calculateXY(SWIPE_DIRECTION.left.end, percentage)
+    )
+  }
+
+  /**
+   * Swipe right based on a percentage
+   */
+  static async swipeRight(percentage = 1): Promise<void> {
+    await this.swipeOnPercentage(
+      this.calculateXY(SWIPE_DIRECTION.right.start, percentage),
+      this.calculateXY(SWIPE_DIRECTION.right.end, percentage)
+    )
+  }
+
+  /**
+   * Swipe from coordinates (from) to the new coordinates (to). The given coordinates are
+   * percentages of the screen.
+   */
+  static async swipeOnPercentage(from: XY, to: XY): Promise<void> {
+    // Get the screen size and store it so it can be re-used.
+    // This will save a lot of webdriver calls if this methods is used multiple times.
+    if (!SCREEN_SIZE) {
+      SCREEN_SIZE = await driver.getWindowRect()
+    }
+
+    // Get the start position on the screen for the swipe
+    const pressOptions = this.getDeviceScreenCoordinates(SCREEN_SIZE, from)
+    // Get the move to position on the screen for the swipe
+    const moveToScreenCoordinates = this.getDeviceScreenCoordinates(SCREEN_SIZE, to)
+
+    await this.swipe(pressOptions, moveToScreenCoordinates)
+  }
+
+  /**
+   * Swipe from coordinates (from) to the new coordinates (to). The given coordinates are in pixels.
+   */
+  static async swipe(from: XY, to: XY): Promise<void> {
```

Comment:
You could use JSDoc params to document the paramters: https://jsdoc.app/tags-param.html